src/kleinanzeigen_bot/model/ad_model.py

```
@@ -183,3 +270,18 @@ class Ad(AdPartial):
+
+    @model_validator(mode = "after")
+    def _validate_auto_price_config(self) -> "Ad":
+        # Note: This validation duplicates checks from AdPartial._validate_price_and_price_type
+        # This is intentional: AdPartial validates raw YAML (with optional None values),
+        # while this validator ensures the final Ad object (after merging defaults) is valid
+        if self.auto_price_reduction.enabled:
+            # Type narrowing: AutoPriceReductionConfig validator ensures min_price is not None when enabled
+            # AdPartial validator ensures price is not None when auto_price_reduction is enabled
+            if self.price is not None and self.auto_price_reduction.min_price is not None:
+                if self.auto_price_reduction.min_price > self.price:
+                    raise ValueError("min_price must not exceed price")
```

Ruff (0.14.5)

Avoid specifying long messages outside the exception class

(TRY003)